app/src/main/java/com/peluso/walletguru/model/Account.kt

```diff
@@ -6,22 +6,39 @@ import com.kirkbushman.araw.models.Submission
  * Class which represents a financial account to be tracked in our application. An account is linked
  * to certain subreddits, and based on the performance of the account these posts are prioritized
  */
-class Account(val type: AccountType, val currentBalance: Float) {
+class Account(val type: AccountType, val currentBalance: Float, val percentageChange: Float) {
 
     //TODO: make this able to be stored in a sqlite database or in sharedprefs so we can save a user's
     // balance over time
 
+
     companion object {
-        fun List<Account>.orderSubmissions(submissions: List<Submission>): List<Submission> {
-            //TODO: write the algorithm for how to order a list of submissions based on the
-            // performance of the account
-            return submissions
+        fun List<Account>.orderSubmissions(submissions: List<Submission>, accounts: List<Account>): List<Submission> {
+
+            //list of accounts sorted in order of decreasing percentage change
+            val sortedAccounts = accounts.sortedByDescending { it.percentageChange }
+
+            //map ranked accounts index to account type
+            val subredditRankings = sortedAccounts.map { it.type.toString() to sortedAccounts.indexOf(it) }.toMap()
+
+            //compator using indexes of subreddits in the map
+            val comparator = Comparator<Submission> {submission1, submission2
+            ->
+                when {
+                    subredditRankings.getValue(submission1.subreddit) > subredditRankings.getValue(submission2.subreddit) -> 1
```

Comment:
submissions do not have access to the account type, so this line actually throws an error because none of the subreddits are indices, the account type enumerator is.